src/effects/backends/effectmanifestparameter.h

```diff
@@ -36,14 +36,158 @@ class EffectManifestParameter {
+        KiloHertz,
+        Kilometer,
+        Meter,
+        MegaHertz,
+        Midinote,
+        Mile,
+        Minute,
+        Millmeter,
+        Millisecond,
+        Octave,
+        Percentage,
         /// Fraction of the Sample Rate
         SampleRate,
-        /// Multiples of a Beat
-        Beats,
+        Seconds,
+        Semitone12tet,
+        Time, // units?
     };
 
+    static UnitsHint lv2UnitToUnitsHint(const QString& lv2Unit) {
+        // Add custom LV2 units here (with correct case) and also
+        // in unitsHintToString()
+        if (lv2Unit == QLatin1String("bar")) {
+            return UnitsHint::Bar;
+        } else if (lv2Unit == QLatin1String("beat")) {
+            return UnitsHint::Beat;
+        } else if (lv2Unit == QLatin1String("bpm")) {
+            return UnitsHint::BPM;
+        } else if (lv2Unit == QLatin1String("cent")) {
+            return UnitsHint::Cent;
+        } else if (lv2Unit == QLatin1String("cm")) {
+            return UnitsHint::Centimetre;
+        } else if (lv2Unit == QLatin1String("coef")) {
+            return UnitsHint::Coefficient;
+        } else if (lv2Unit == QLatin1String("db")) {
+            return UnitsHint::Decibel;
+        } else if (lv2Unit == QLatin1String("degree")) {
+            return UnitsHint::Degree;
+        } else if (lv2Unit == QLatin1String("frame")) {
+            return UnitsHint::Frame;
+        } else if (lv2Unit == QLatin1String("hz")) {
+            return UnitsHint::Hertz;
+        } else if (lv2Unit == QLatin1String("inch")) {
+            return UnitsHint::Inch;
+        } else if (lv2Unit == QLatin1String("khz")) {
+            return UnitsHint::KiloHertz;
+        } else if (lv2Unit == QLatin1String("km")) {
+            return UnitsHint::Kilometer;
+        } else if (lv2Unit == QLatin1String("m")) {
+            return UnitsHint::Meter;
+        } else if (lv2Unit == QLatin1String("mhz")) {
+            return UnitsHint::MegaHertz;
+        } else if (lv2Unit == QLatin1String("midiNote")) {
+            return UnitsHint::Midinote;
+        } else if (lv2Unit == QLatin1String("mile")) {
+            return UnitsHint::Mile;
+        } else if (lv2Unit == QLatin1String("min")) {
+            return UnitsHint::Minute;
+        } else if (lv2Unit == QLatin1String("mm")) {
+            return UnitsHint::Millmeter;
+        } else if (lv2Unit == QLatin1String("ms")) {
+            return UnitsHint::Millisecond;
+        } else if (lv2Unit == QLatin1String("oct")) {
+            return UnitsHint::Octave;
+        } else if (lv2Unit == QLatin1String("pc")) {
+            return UnitsHint::Percentage;
+        } else if (lv2Unit == QLatin1String("s")) {
+            return UnitsHint::Seconds;
+        } else if (lv2Unit == QLatin1String("semitone12TET")) {
+            return UnitsHint::Semitone12tet;
+        } else {
+            return UnitsHint::Unknown;
+        }
+    }
+
+    static QString unitsHintToString(const UnitsHint unitsHint) {
```

Comment:
This table can be changed to return the format string (or LB2 renderer).
Like "%f ms"